driver/src/test/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraphIndexTests.java

```diff
@@ -0,0 +1,77 @@
+package org.apache.tinkerpop.gremlin.orientdb;
+
+import com.orientechnologies.orient.core.metadata.schema.OType;
+import com.orientechnologies.orient.core.storage.ORecordDuplicatedException;
+import org.apache.commons.configuration.BaseConfiguration;
+import org.apache.commons.configuration.Configuration;
+import org.apache.tinkerpop.gremlin.process.traversal.P;
+import org.apache.tinkerpop.gremlin.structure.T;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.*;
+
+public class OrientGraphIndexTests {
+
+    public static final String URL = "memory:" + OrientGraphIndexTests.class.getSimpleName();
+//    public static final String URL = "remote:localhost/test";
+
+    private OrientGraph newGraph() {
+        return new OrientGraphFactory(URL + UUID.randomUUID(), "root", "root").getNoTx();
+    }
+
+    @Test
+    public void uniqueIndex() {
+        String className = "V";
+        String keyName = "key1";
+        String value = "value1";
+
+        OrientGraph graph = newGraph();
+        graph.createVertexClass(className);
+
+        Configuration config = new BaseConfiguration();
+//        config.setProperty("class", className);
+        config.setProperty("type", "UNIQUE");
+        config.setProperty("keytype", OType.STRING);
+        graph.createIndex(keyName, Vertex.class, config);
+
+        Assert.assertEquals(graph.getIndexedKeys(Vertex.class), new HashSet<String>(Collections.singletonList(keyName)));
+
+        graph.addVertex(T.label, className, keyName, value);
+
+        // This test doesn't check that the following traversal hit the index.
+        // Only verified by println debugging in OrientGraphStep.
```

Comment:
I suppose that there should be a test for the traversals use of indexing in a `OrientGraphStepTest` suite. Haven't figured out a way to do that yet. It seems that TinkerGraph doesn't test it either.